test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java

```diff
@@ -708,11 +708,14 @@ protected void refreshAllIndices() throws IOException {
         requestOptions.setWarningsHandler(warnings -> {
             if (warnings.isEmpty()) {
                 return false;
-            } else if (warnings.size() > 1) {
-                return true;
-            } else {
-                return warnings.get(0).startsWith("this request accesses system indices:") == false;
             }
+            boolean allSystemIndexWarning = true;
+            for (String warning : warnings) {
+                if (!warning.startsWith("this request accesses system indices:")) {
+                    allSystemIndexWarning = false;
```

Comment:
Minor, we could break the loop right away:

```suggestion
                    allSystemIndexWarning = false;
                    break;
```

Reply:
Added